components/app-framework/src/widgets/file-drop/item-drop-helper.service.js

```diff
@@ -0,0 +1,254 @@
+(function () {
+  'use strict';
+
+  angular
+    .module('app.framework.widgets')
+    .factory('itemDropHelper', itemDropHelper);
+
+  function itemDropHelper($q, gitIgnoreParser) {
+
+    var archiveRegex = /\.(tar|zip|tar.gz)$/i;
+
+    return {
```

Comment:
Possibly - I have updated all relevant places to add handlers that reject the promise.